sitegeist/sitegeist-neos-base-distribution
# FEATURE: Integrate TestCafé

With this PR, a TestCafé integration comes to our base distribution. E2E tests can then be executed via `make test-e2e`.

## WARNING!

This PR depends on https://github.com/sitegeist/docker-neos/pull/2. As long as it is not merged and published on docker hub, the TestCafé  image must be built locally and tagged as `sitegeist/neos-testcafe`.

You can do this like so:
```
git clone git@github.com:sitegeist/docker-neos.git
cd docker-neos
git checkout feature/testcafe-image
cd src/testcafe
docker build -t sitegeist/neos-testcafe .
```

After the above image has been published, don't forget to remove your local build via:
```
docker rmi sitegeist/neos-testcafe
```

## Steps to test

1. Clone the repository and set up the distribution
```
git clone git@github.com:sitegeist/sitegeist-neos-base-distribution.git
cd sitegeist-neos-base-distribution
make up
make install
```

2. Enter the php-fpm container
```
make ssh
```

3. Execute all DB migrations and import the placeholder site package within the php-fpm container
```
./flow doctrine:migrate
./flow site:import --package-key Sitegeist.Site.Placeholder
exit
```

4. Run the E2E tests outside the php-fpm container
```
make test-e2e
```

This takes a little while. Afterwards your output should be:

```
 Running tests in:
 - Chrome 71.0.3578 / Linux 0.0.0

 #00000 - Placeholder Test
 ✓ Site Header is present
 ✓ Site Footer is present


 2 passed (1s)
```
